src/components/Staff/UpdateStaffForTimeRangeForm.tsx

```diff
@@ -46,18 +46,25 @@ export const UpdateStaffForTimeRangeForm = ({
     setEndDate(date?.startOf('day') || moment());
   };
 
-  const handleStartTimeChange = (date: Moment | null) => {
-    setStartTime(date?.startOf('minute') || moment());
-    if (date && endTime && date.isAfter(endTime)) {
+  const handleStartTimeChange = (hour: number, minute: number) => {
+    const newStartTime = moment(startTime).set({hour, minute});
+    setStartTime(newStartTime);
+    if (newStartTime.isAfter(endTime)) {
       setError("Start time must be less than or equal to end time.");
       return;
     }
     setError(null);
   };
 
-  const handleEndTimeChange = (date: Moment | null) => {
-    setEndTime(date?.startOf('minute') || moment());
-    if (date && startTime && date.isBefore(startTime)) {
+  const handleEndTimeChange = (hour: number, minute: number) => {
+    let newEndTime: Moment;
+    if (startTime.hour() === 0 && startTime.minutes() === 0 && hour === 0 && minute === 0) {
+      newEndTime = moment(startTime).add(1, 'day').startOf('day');
+    } else {
+      newEndTime = moment(startTime).set({hour, minute});
+    }
+    setEndTime(newEndTime);
+    if (newEndTime.isBefore(startTime)) {
```

Comment:
This needs to be before or equal. It's not valid to choose the same start and end time